api/controller/caasapplicationprovisioner/client.go

```diff
@@ -442,3 +416,67 @@ func (c *Client) RemoveUnit(unitName string) error {
 	}
 	return resultErr
 }
+
+// DestroyUnits is responsible for starting the process of destroying units
+// associated with this application.
+func (c *Client) DestroyUnits(unitNames []string) error {
+	args := params.DestroyUnitsParams{}
+	args.Units = make([]params.DestroyUnitParams, 0, len(unitNames))
+
+	for _, unitName := range unitNames {
+		tag := names.NewUnitTag(unitName)
+		args.Units = append(args.Units, params.DestroyUnitParams{
+			UnitTag: tag.String(),
+		})
+	}
+	result := params.DestroyUnitResults{}
+
+	err := c.facade.FacadeCall("DestroyUnits", args, &result)
+	if err != nil {
+		return err
+	}
+
+	if len(result.Results) != len(unitNames) {
+		return fmt.Errorf("expected %d results got %d", len(unitNames), len(result.Results))
+	}
+
+	for _, res := range result.Results {
+		if res.Error != nil {
+			return errors.Trace(res.Error)
+		}
+	}
+
+	return nil
+}
+
+// ProvisioningState returns the current provisioning state for the CAAS application.
+// The result can be nil.
+func (c *Client) ProvisioningState(appName string) (*params.CAASApplicationProvisioningState, error) {
+	var result params.CAASApplicationProvisioningStateResult
+	args := params.Entity{Tag: names.NewApplicationTag(appName).String()}
+	err := c.facade.FacadeCall("ProvisioningState", args, &result)
+	if err != nil {
+		return nil, err
+	}
+	if result.Error != nil {
+		return nil, result.Error
+	}
+	return result.ProvisioningState, nil
+}
+
+// SetProvisioningState sets the provisioning state for the CAAS application.
+func (c *Client) SetProvisioningState(appName string, state params.CAASApplicationProvisioningState) error {
+	var result params.ErrorResult
+	args := params.CAASApplicationProvisioningStateArg{
+		Application:       params.Entity{Tag: names.NewApplicationTag(appName).String()},
+		ProvisioningState: state,
+	}
+	err := c.facade.FacadeCall("SetProvisioningState", args, &result)
```

Comment:
this is a controller API, so if and when it is needed we can do it. But IMO, this will never need to be a bulk call since all caasapplicationprovisoner worker is likely to stay as seperate workers per application.